src/v/cloud_roles/apply_aws_credentials.cc

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2022 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+
+#include "cloud_roles/apply_aws_credentials.h"
+
+struct aws_header_names {
+    static constexpr boost::beast::string_view x_amz_security_token
+      = "x-amz-security-token";
+    static constexpr boost::beast::string_view x_amz_content_sha256
+      = "x-amz-content-sha256";
+};
+
+/// Prefix used for testing, so that if testing against minio, tokens are
+/// fetched but not injected to header. Minio does not accept random security
+/// tokens with requests.
+static constexpr const char* ci_test_token_prefix = "__REDPANDA_SKIP_IAM_TOKEN";
+
+static bool is_test_token(std::string_view token) {
+    return token.find(ci_test_token_prefix) == 0;
+}
+
+struct aws_signatures {
+    /// SHA256 of an empty string (used in situation when the signed payload is
+    /// has zero length)
+    static constexpr std::string_view emptysig
+      = "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855";
+
+    static constexpr std::string_view unsigned_payload = "UNSIGNED-PAYLOAD";
+};
+
+cloud_roles::apply_aws_credentials::apply_aws_credentials(
+  cloud_roles::aws_credentials&& credentials)
+  : _signature{credentials.region, credentials.access_key_id, credentials.secret_access_key}
+  , _session_token{credentials.session_token} {}
+
+std::error_code cloud_roles::apply_aws_credentials::add_auth(
+  http::client::request_header& header) const {
+    if (_session_token) {
+        auto st = *_session_token;
+        if (!is_test_token(st())) {
```

Comment:
I dislike this approach too generally but I couldn't find a way around it. 

If the token is sent to minio it rejects the request with invalid token error. If the token is missing in sts response we (correctly) fail with error. 

I couldn't figure out a way not to send the token in a test context to minio other than this.

Reply:
you might consider basing this on `getenv` so that nothing is hard coded (just in case) and it can then be modified when starting redpanda in ducktape by passing in different environment variable. I think that @jcsp did something similar a few times. not sure if it offers a strong advantage over this, but it is nice to have that security of avoiding hard coded special keys.

Reply:
So IIUC we have a special env variable which if present signals that the token should be skipped? We can do that, we do something similar to setup test ip address and port in this PR for testing remote APIs.